endpoints/openrtb2/video_auction.go

```diff
@@ -0,0 +1,528 @@
+			labels.CookieFlag = pbsmetrics.CookieFlagNo
+		} else {
+			labels.CookieFlag = pbsmetrics.CookieFlagYes
+		}
+	}
+
+	numImps = len(bidReq.Imp)
+
+	//execute auction logic
+	response, err := deps.ex.HoldAuction(ctx, bidReq, usersyncs, labels, &deps.categories)
+	ao.Request = bidReq
+	ao.Response = response
+	if err != nil {
+		errL := []error{err}
+		handleError(labels, w, errL, ao)
+		return
+	}
+
+	//build simplified response
+	bidResp, err := buildVideoResponse(response)
+	if err != nil {
+		errL := []error{err}
+		handleError(labels, w, errL, ao)
+		return
+	}
+	if bidReq.Test == 1 {
+		bidResp.Ext = response.Ext
+	}
+
+	resp, err := json.Marshal(bidResp)
+	//resp, err := json.Marshal(response)
+	if err != nil {
+		errL := []error{err}
+		handleError(labels, w, errL, ao)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(resp)
+
+}
+
+func handleError(labels pbsmetrics.Labels, w http.ResponseWriter, errL []error, ao analytics.AuctionObject) {
+	labels.RequestStatus = pbsmetrics.RequestStatusErr
+	w.WriteHeader(http.StatusInternalServerError)
+	var errors string
+	for _, er := range errL {
+		errors = fmt.Sprintf("%s %s", errors, er.Error())
+	}
+	fmt.Fprintf(w, "Critical error while running the video endpoint: %v", errors)
+	glog.Errorf("/openrtb2/video Critical error: %v", errors)
+	ao.Status = http.StatusInternalServerError
+	ao.Errors = append(ao.Errors, errL...)
+}
+
+func (deps *endpointDeps) createImpressions(videoReq *openrtb_ext.BidRequestVideo) (imps []openrtb.Imp, errL []error) {
+	videoDur := videoReq.PodConfig.DurationRangeSec
+	minDuration, maxDuration := minMax(videoDur)
+	reqExactDur := videoReq.PodConfig.RequireExactDuration
+	videoData := videoReq.Video
+
+	finalImpsArray := make([]openrtb.Imp, 0)
+	for _, pod := range videoReq.PodConfig.Pods {
+
+		//load stored impression
+		storedImpressionId := string(pod.ConfigId)
+		storedImp, errs := deps.loadStoredImp(storedImpressionId)
+		if errs != nil {
+			return nil, errs
+		}
+
+		numImps := pod.AdPodDurationSec / minDuration
+
+		impDivNumber := numImps / len(videoDur)
+		if impDivNumber == 0 && reqExactDur {
+			err := errors.New("Invalid request: unable to build impressions with given duration array size")
+			errL := []error{err}
+			return nil, errL
+		}
+
+		impsArray := make([]openrtb.Imp, numImps)
+		for impInd := range impsArray {
+			newImp := createImpressionTemplate(storedImp, videoData)
+			impsArray[impInd] = newImp
+			if reqExactDur {
+				//floor := int(math.Floor(ind/impDivNumber))
+				durationIndex := impInd / impDivNumber
+				if durationIndex > len(videoDur)-1 {
+					durationIndex = len(videoDur) - 1
+				}
+				impsArray[impInd].Video.MaxDuration = int64(videoDur[durationIndex])
+				impsArray[impInd].Video.MinDuration = int64(videoDur[durationIndex])
+				//fmt.Println(podIndex, "  ", impInd, "duration ", videoDur[durationIndex])
+			} else {
+				impsArray[impInd].Video.MaxDuration = int64(maxDuration)
```

Comment:
Should we also set `MinDuration` here?

Reply:
As I understand no, this is a case for `requeredExactDuration==false`, `minDuration` is 0 by default

Reply:
Yes. The min duration should be set to default 0. This is in sync with the product approved behavior for client side sahara.